src/test/java/dev/jbang/idea/run/JavaSnippetLineMarkerProvider.kt

```diff
@@ -0,0 +1,157 @@
+package dev.jbang.idea.run
+
+import com.intellij.codeInsight.daemon.LineMarkerInfo
+import com.intellij.execution.ExecutionException
+import com.intellij.execution.Executor
+import com.intellij.execution.configurations.GeneralCommandLine
+import com.intellij.execution.configurations.PtyCommandLine
+import com.intellij.execution.executors.DefaultRunExecutor
+import com.intellij.execution.lineMarker.RunLineMarkerProvider
+import com.intellij.execution.runners.ExecutionEnvironmentBuilder
+import com.intellij.ide.IdeBundle
+import com.intellij.ide.actions.runAnything.commands.RunAnythingCommandCustomizer
+import com.intellij.ide.actions.runAnything.execution.RunAnythingRunProfile
+import com.intellij.openapi.actionSystem.DataContext
+import com.intellij.openapi.actionSystem.impl.SimpleDataContext
+import com.intellij.openapi.editor.markup.GutterIconRenderer
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.project.guessProjectDir
+import com.intellij.openapi.ui.Messages
+import com.intellij.openapi.util.registry.Registry
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiElement
+import com.intellij.psi.javadoc.PsiSnippetAttributeList
+import com.intellij.psi.javadoc.PsiSnippetDocTag
+import com.intellij.psi.javadoc.PsiSnippetDocTagBody
+import com.intellij.psi.javadoc.PsiSnippetDocTagValue
+import com.intellij.util.execution.ParametersListUtil
+import dev.jbang.idea.jbangIcon
+import org.jetbrains.kotlin.psi.psiUtil.getChildOfType
+import java.io.File
+import javax.swing.Icon
+
+@Suppress("UnstableApiUsage")
+class JavaSnippetLineMarkerProvider : RunLineMarkerProvider() {
```

Comment:
this shouldn't be in the test package should it?

p.s. would be great to add tests for new features

Reply:
I just use it for local development and test, and plugin development env has been moved to IntelliJ IDEA 2022.1 already.   

```
platformType = IC
platformVersion = 221.5080.210
```

I can not put JavaSnippetLineMarkerProvider.kt under `src/main` because of plugin verification.  Plugin verifier will check every class in the plugin, and this class can make verification failed.